examples/tutorials/advanced/legends.py

```diff
@@ -0,0 +1,159 @@
+# -------------------
+#
+# Use the ``position`` parameter to adjust the position of the legend. Add an offset via
+# **+o** for the x and y directions. Additionally append **+w** to adjust the ``width``
+# of the length. Note, no box is drawn by default if ``position`` is used.
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+fig.plot(x=0, y=0, style="c0.25c", fill="orange", label="orange circle")
+fig.plot(x=1, y=0, style="t0.3c", fill="pink", pen="black", label="pink triangle")
+fig.plot(x=[-3, 3], y=[-2, -2], pen="darkred", label="darkred line")
+
+# Set the reference point to the Top Left corner within (lowercase "j") the bounding box
+# of the plot and use offsets of 0.3 and 0.2 centimeters in the x and y directions,
+# respectively.
+fig.legend(position="jTL+o0.3c/0.2c")
+
+fig.show()
+
+
+# %%
+# Add a box
+# ---------
+# Use the ``box`` parameter for adjusting the box around the legend. The outline of the
+# box can be adjusted by appending **+p**. Append **+g** to fill the legend with a color
+# (or pattern) [Default is no fill]. The default of ``position`` is preserved.
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame="rltb+glightgray")
+
+fig.plot(x=0, y=0, style="c0.25c", fill="orange", label="orange circle")
+fig.plot(x=1, y=0, style="t0.3c", fill="pink", pen="black", label="pink triangle")
+fig.plot(x=[-3, 3], y=[-2, -2], pen="darkred", label="darkred line")
+
+fig.legend(position="jTL+o0.3c/0.2c", box=True)
+
+fig.shift_origin(xshift="w+1c")
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame="rltb+glightgray")
+
+fig.plot(x=0, y=0, style="c0.25c", fill="orange", label="orange circle")
+fig.plot(x=1, y=0, style="t0.3c", fill="pink", pen="black", label="pink triangle")
+fig.plot(x=[-3, 3], y=[-2, -2], pen="darkred", label="darkred line")
+
+# Add a box with a 2-points thick blue, solid outline and a white fill with a
+# transparency of 70 percentage ("@30").
+fig.legend(position="jTL+o0.3c/0.2c", box="+p2p,blue+gwhite@30")
+
+fig.show()
+
+
+# %%
+# Create a manual legend
+# ----------------------
+#
+# For more complicated legends, users need to prepare a legend specification with
+# instructions for the layout of the legend entries. In PyGMT, the legend specification
+# can be either an ASCII file or an :class:`io.StringIO` object. Both are passed to the
+# ``spec`` parameter of :meth:`pygmt.Figure.legend`. Multiple legend codes are available
+# to create complicated legends. In the example below we show an subset; a full
+# overview can be found at :gmt-docs:`legend.html#legend-codes`. It's also supported to
+# include length scales (for geographic projections), faults, and images as well as to
+# add specific lines.
+#
+# The following example is orientated on the related GMT example at
+# :gmt-docs:`legend.html#examples`, but modified to use an :class:`io.StringIO` object.
+#
+# We start with setting up the :class:`io.StringIO` object.
```

Comment:
I am a bit unsure how detailed we want to explain the syntax of the legend specification here.

Reply:
I don't think we should explain the details here. We should refer users to read the GMT legend manual, or if we really want to explain it, it should be in the Technical Referecence, rather than in a tutorial.

Reply:
Agree, I also do not think it's good to have the details explained here; having this listed in the Technical Reference is a good point!